addons/point_of_sale/static/src/js/Screens/ClientListScreen/ClientListScreen.js

```diff
@@ -173,6 +179,34 @@ odoo.define('point_of_sale.ClientListScreen', function(require) {
         cancelEdit() {
             this.deactivateEditMode();
         }
+        async searchClient() {
+            let result = await this.getNewClient();
+            this.env.pos.db.add_partners(result);
+            if(!result.length) {
+                await this.showPopup('ErrorPopup', {
+                    title: this.env._t(''),
+                    body: this.env._t('No customer found'),
+                });
+            }
+            this.render();
+        }
+        async getNewClient(domain) {
```

Comment:
You are passing a domain here, but it is immediately overridden by empty array, is it really needed? Looks to me that the domain is not needed and this method depends entirely on `this.state.query`.